aspnet/mvc/controllers/areas.rst

```diff
@@ -35,19 +43,63 @@ Instead of having all of the controllers located under the Controllers parent di
 
           - Index.cshtml
 
-Looking at the preceding directory hierarchy example, there are a few guidelines to keep in mind when defining areas:
+When MVC tries to render a view in an Area, by default, it tries to look in the following locations:
+
+.. code-block:: none
+
+  /Areas/<Area-Name>/Views/<Controller-Name>/<Action-Name>.cshtml
+  /Areas/<Area-Name>/Views/Shared/<Action-Name>.cshtml
+  /Views/Shared/<Action-Name>.cshtml
+
+These are the default locations which can be changed via the ``AreaViewLocationFormats`` on the ``Microsoft.AspNetCore.Mvc.Razor.RazorViewEngineOptions``.
 
-- A directory called *Areas* must exist as a child directory of the project.
-- The *Areas* directory contains a subdirectory for each of your project's areas (*Products* and *Services*, in this example).
-- Your controllers should be located as follows:
-  ``/Areas/[area]/Controllers/[controller].cs``
-- Your views should be located as follows:
-  ``/Areas/[area]/Views/[controller]/[action].cshtml``
+For example, in the below code instead of having the folder name as 'Areas', it has been changed to 'Categories'.
 
-Note that if you have a view that is shared across controllers, it can be located in either of the following locations:
+.. code-block:: c#
+
+  services.Configure<RazorViewEngineOptions>(options =>
+  {
+      options.AreaViewLocationFormats.Clear();
+      options.AreaViewLocationFormats.Add("/Categories/{2}/Views/{1}/{0}.cshtml");
+      options.AreaViewLocationFormats.Add("/Categories/{2}/Views/Shared/{0}.cshtml");
+      options.AreaViewLocationFormats.Add("/Views/Shared/{0}.cshtml");
+  });
+
+One thing to note is that the structure of the ``Views`` folder is the only one which is considered important here and the content of the rest of the folders like 'Controllers' and 'Models' does **not** matter. So for example, we could change the above folder structure to be like the following and the application would still work as before.
+
+- Project name
 
-- ``/Areas/[area]/Views/Shared/[action].cshtml``
-- ``/Views/Shared/[action].cshtml``
+  - Areas
+
+    - Products
+
+      - HomeController.cs
+
+      - ManageController.cs
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+        - Manage
+
+          - Index.cshtml
+
+    - Services
+
+      - HomeController.cs
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+This works because the content of 'Controllers' and 'Models' is just code which gets compiled into a .dll where as the content of the 'Views' is not until a request to that view has been made.  
+
+Even though the above folder structure works, this is not typical and you would want to structure your code as shown in the beginning of this article.
```

Comment:
Do we want to show folks an `atypical` example? Maybe having the first structure and leaving the note in might be less confusing?